Monika After Story/game/event-rules.rpy

```diff
@@ -879,11 +877,24 @@ init python:
             IN:
                 per_rules - persistent dict/list to get rules from
             """
+            remove_list = list()
+
             for ev_label in per_rules:
                 ev = mas_getEV(ev_label)
-                if ev is not None and MASUndoActionRule.evaluate_rule(ev):
+                #Since we can have differing returns, we store this to use later
+                should_undo = MASUndoActionRule.evaluate_rule(ev)
+
+                #If we do have the dates and we're out of the time period, we undo the action
+                if ev is not None and should_undo:
                     Event._undoEVAction(ev)
 
+                #We ended up getting a none, so we need to add the ev to a list of ons we need to remove
+                elif should_undo is None:
+                    remove_list.append(ev)
+
+            #Now we go thru the items in remove list and remove their rules
+            for ev in remove_list:
+                MASUndoActionRule.remove_rule(ev)

```

Comment:
`check_persistent_rules` takes a dict of rules to evaluate. However, `MASUndoActionRule.remove_rule` removes rules from the persistent dict. This is bad if we wanted to evaluate and clean an arbitrary collection of rules not from the persistent dict.

`remove_rule` shouldn't be used here. This should just remove directly from the given dict of rules.